src/oob/xml.rs

```diff
@@ -0,0 +1,190 @@
+//! Support for RPKI XML structures.
+
+use std::{fs, io};
+use std::path::Path;
+use xml::EventReader;
+use xml::ParserConfig;
+use xml::reader::XmlEvent;
+use xml::attribute::OwnedAttribute;
+
+
+//------------ XmlReader -----------------------------------------------------
+
+/// A convenience wrapper for RPKI XML parsing
+///
+/// This type only exposes things we need for the RPKI XML structures.
+pub struct XmlReader<R: io::Read> {
```

Comment:
I’d suggest to rearrange XmlReader in such a way that its access is similar to the way the ber crate allows access, i.e., via nested calls to a `take_element` method that takes a closure and passes on the XmlReader, ensuring correct nesting of XML elements and such.

Reply:
Good suggestion, thanks for the pointers. Done.